lib/redis.rb

```diff
@@ -2803,6 +2803,326 @@ def geodist(key, member1, member2, unit = 'm')
+  #
+  # @return [Integer] the number of entries actually deleted
+  def xtrim(key, maxlen, modifier: false)
+    args = [:xtrim, key, 'MAXLEN', (modifier ? '~' : nil), maxlen].compact
+    synchronize { |client| client.call(args) }
+  end
+
+  # Delete entries by entry ids.
+  #
+  # @example With splatted entry ids
+  #   redis.xdel('mystream', '0-1', '0-2')
+  # @example With arrayed entry ids
+  #   redis.xdel('mystream', ['0-1', '0-2'])
+  #
+  # @param key [String]        the stream key
+  # @param ids [Array<String>] one or multiple entry ids
+  #
+  # @return [Integer] the number of entries actually deleted
+  def xdel(key, *ids)
+    args = [:xdel, key].concat(ids.flatten)
+    synchronize { |client| client.call(args) }
+  end
+
+  # Fetches entries of the stream.
+  #
+  # @example Without options
+  #   redis.xrange('mystream')
+  # @example With first entry id option
+  #   redis.xrange('mystream', first: '0-1')
+  # @example With first and last entry id options
+  #   redis.xrange('mystream', first: '0-1', last: '0-3')
+  # @example With count options
+  #   redis.xrange('mystream', count: 10)
+  #
+  # @param key   [String]  the stream key
+  # @param first [String]  first entry id of range, default value is `-`
+  # @param last  [String]  last entry id of range, default value is `+`
+  # @param count [Integer] the number of entries as limit
+  #
+  # @return [Hash{String => Hash}] entries
+  def xrange(key, first: '-', last: '+', count: nil)
+    args = [:xrange, key, first, last]
+    args.concat(['COUNT', count]) unless count.nil?
+    synchronize { |client| client.call(args, &HashifyStreamEntries) }
+  end
+
+  # Fetches entries of the stream ordered by descending.
+  #
+  # @example Without options
+  #   redis.xrevrange('mystream')
+  # @example With first entry id option
+  #   redis.xrevrange('mystream', first: '0-1')
+  # @example With first and last entry id options
+  #   redis.xrevrange('mystream', first: '0-1', last: '0-3')
+  # @example With count options
+  #   redis.xrevrange('mystream', count: 10)
+  #
+  # @param key   [String]  the stream key
+  # @param first [String]  first entry id of range, default value is `-`
+  # @param last  [String]  last entry id of range, default value is `+`
+  # @param count [Integer] the number of entries as limit
+  #
+  # @return [Hash] entries
+  def xrevrange(key, first: '-', last: '+', count: nil)
+    args = [:xrevrange, key, last, first]
+    args.concat(['COUNT', count]) unless count.nil?
+    synchronize { |client| client.call(args, &HashifyStreamEntries) }
+  end
+
+  # Returns the number of entries inside a stream.
+  #
+  # @example With key
+  #   redis.xlen('mystream')
+  #
+  # @param key [String] the stream key
+  #
+  # @return [Integer] the number of entries
+  def xlen(key)
+    synchronize { |client| client.call([:xlen, key]) }
+  end
+
+  # Fetches entries from one or multiple streams. You're able to use blocking if needed.
```

Comment:
`You're able to use blocking if needed.` I think this needs to be phrased differently.

Maybe `Optionally blocking` ?